framework/src/controller/action.ts

```diff
@@ -32,56 +33,72 @@ export interface ActionsObject {
 	[key: string]: Action;
 }
 
+type ID = string | number | null;
+
 export class Action {
-	public module: string;
-	public name: string;
-	public handler?: (action: ActionInfoObject) => unknown;
-	public source?: string;
+	public jsonrpc = VERSION;
+	public id: ID;
+	public method: string;
 	public params: object;
+	public source?: string;
+	public handler?: (action: ActionInfoObject) => unknown;
 
 	public constructor(
-		name: string,
+		id: ID,
+		method: string,
 		params?: object,
 		source?: string,
 		handler?: (action: ActionInfoObject) => unknown,
 	) {
 		assert(
-			actionWithModuleNameReg.test(name),
-			`Action name "${name}" must be a valid name with module name.`,
+			actionWithModuleNameReg.test(method),
+			`Action method "${method}" must be a valid method with module name and action name.`,
 		);
-		[this.module, this.name] = name.split(':');
-		this.params = params ?? {};
-
 		if (source) {
 			assert(moduleNameReg.test(source), `Source name "${source}" must be a valid module name.`);
 			this.source = source;
 		}
 
+		this.id = id;
```

Comment:
This attribute seems required but not asserted neither any default value is assigned.